manifests-gen/providers.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	configclient "sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/repository"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/yamlprocessor"
+	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	powerVSProvider         = "powervs"
+	ibmCloudProvider        = "ibmcloud"
+	metadataFilePath        = "./metadata.yaml"
+	kustomizeComponentsPath = "./config/default"
+)
+
+type provider struct {
+	Name       string                    `json:"name"`
+	PType      clusterctlv1.ProviderType `json:"type"`
+	Version    string                    `json:"version"`
+	components repository.Components
+	metadata   []byte
+}
+
+// loadComponents loads components from the given provider.
+func (p *provider) loadComponents() error {
+	// Create new clusterctl config client
+	configClient, err := configclient.New("")
+	if err != nil {
+		return err
+	}
+
+	// Create new clusterctl provider client
+	providerConfig, err := configClient.Providers().Get(p.Name, p.PType)
+	if err != nil {
+		return err
+	}
+
+	// Set options for yaml processor
+	options := repository.ComponentsOptions{
+		TargetNamespace:     targetNamespace,
+		SkipTemplateProcess: true,
+	}
+
+	// Compile assets using kustomize.
+	rawComponents, err := fetchAndCompileComponents(path.Join(projDir, kustomizeComponentsPath))
+	if err != nil {
+		return err
+	}
+
+	// Ininitialize new clusterctl repository components, this should some yaml processing
+	p.components, err = repository.NewComponents(repository.ComponentsInput{
+		Provider:     providerConfig,
+		ConfigClient: configClient,
+		Processor:    yamlprocessor.NewSimpleProcessor(),
+		RawYaml:      rawComponents,
+		Options:      options,
+	})
+	if err != nil {
+		return err
+	}
+
+	content, err := os.ReadFile(path.Join(projDir, metadataFilePath))
+	if err != nil {
+		return fmt.Errorf("error while reading metadata file: %w", err)
+	}
+
+	p.metadata = content
+
+	return err
+}
+
+func (p *provider) providerTypeName() string {
+	return strings.ReplaceAll(strings.ToLower(string(p.PType)), "provider", "")
+}
+
+func (p *provider) writeProviderComponentsConfigmap(fileName string, objs []unstructured.Unstructured) error {
+	combined, err := utilyaml.FromUnstructured(objs)
+	if err != nil {
+		return err
+	}
+
+	annotations := openshifAnnotations
```

Comment:
Typo

Reply:
Same as https://github.com/openshift/cluster-capi-operator/pull/121#discussion_r1405973164

Reply:
Anyway, I can fix the typo